### ISSUE TITLE
highstate changes spurious warnings

### ISSUE THREAD

**Issue by kiorky** [saltstack/salt#21120]
At first, @s0undt3ch & @thatch45 i would be happy that the highstate behavior stay in the form as @s0undt3ch just commited, with the warnings summary, it's awsome.

Now, i have spurious warnings, from time to time, where i dont have those 'fun' & 'state' kwargs, i dont understand from where they are injected and throw the warnings:

```
ID: npm-version-0_10_36auto
Function: archive.extracted
Name: /srv/apps/node/0.10.36
Result: True
Comment: /srv/apps/node/0.10.36/bin/.node_0.10.36 already exists
Started: 17:54:00.978127
Duration: 0.643 ms
Changes:   
Warnings: 'fun' and 'state' are invalid keyword arguments for
          'archive.extracted'. If you were trying to pass additional data to
          be used in a template context, please populate 'context' with
          'key: value' pairs. Your approach will work until Salt Carbon is
          out. Please update your state files.
```

**Comment by kiorky** [comment]
(in other terms, never hardfail like @s0undt3ch changed two times, or we can maybe add a config switch to disable hardfail and get that warning summary in this case)

**Comment by s0undt3ch** [comment]
The problem with the hard failure, which will happen again in Salt Carbon, is that, either I forgot to include the warnings when this was initially implemented(I added a test case but was only checking the `ret` not the outputters), or the hightstate outputter changed and started to ignore it(since it wasn't explicitly handled). My guess is the first choice.

I also added a test case to make sure the warnings are shown in the outputters.

Thanks for reporting this, I'll track down where that warning is coming from since those kwargs are supposed not to trigger the warnings.

If you find any more of these, please report them.

Thanks!

**Comment by kiorky** [comment]
@s0undt3ch but why going to be hard, on carbon, afterall, why ?
This sounds like an orangina pub, but really, we could be more nice and leave as it is done right now, could'nt  we ? And then removing the ''' Your approach will work until Salt Carbon is
          out. Please update your state files.''' stanza

**Comment by s0undt3ch** [comment]
Can you please also paste a sanitized version of the state file that triggered this?

**Comment by kiorky** [comment]
ha.
this is from the makina-states.mc_states.mc_project machinery, so not easy in bare salt, but i have an idea on how to reproduce it. lemme sketch it.

**Comment by s0undt3ch** [comment]
The hard failure is to enforce proper states while users are still writing them. The reason is that we've had a few bugs reported which weren't bugs, it was bad state files, with wrong or mistyped arguments.

With this change, users will know what's happening instead of thinking that everything is right when its not.

Don't make this a bad thing. If you pass a wrong keyword argument to a python function, what happens? Python complains. This is the exact same thing. And Carbon is 3 major versions from now which gives a fair amount of time for users to update their states with proper and correct usage....

**Comment by kiorky** [comment]
Then it is a too major behavior change, it would be a disaster for us now, a config opt would make me happy even if the behavior is hardcrash as a default, if i can lower it down to warning, it's ok.

**Comment by kiorky** [comment]
To reproduce
```
testb:
  file.absent:
    - name: /tmp/foononfoo

```

**Comment by s0undt3ch** [comment]
Because its a major change we'll take 3 major versions to fully deprecate it...

**Comment by s0undt3ch** [comment]
Oh, thanks for the SLS, I'll track it down...

**Comment by kiorky** [comment]
My users wont change their states, they mostly dont understand them ... :)

**Comment by kiorky** [comment]
The warnings will make them act, the hardfail with loose them.

**Comment by kiorky** [comment]
Im not at all the hardfail by default, just add a test from __opts__ that crashes instead of warning, that's not a big deal and everyone will be happy.

**Comment by s0undt3ch** [comment]
You run your states through highstate?

**Comment by kiorky** [comment]
Nope, sls

**Comment by kiorky** [comment]
On that node, the cmd i used was:
```
salt-call -lall  --local state.sls s21120

```

**Comment by kiorky** [comment]
```
default_include: minion.d/*.conf
master: 0.0.0.1
ipv6: False
retry_dns: 30
master_port: 4506
user: root
pidfile: /var/run/salt-minion.pid
root_dir: /
pki_dir: /etc/salt/pki/minion
id: prod-rempeitc.makina-corpus.net
cachedir: /var/cache/salt/salt-minion
verify_env: True
cache_jobs: False
sock_dir: /var/run/salt/minion
color: True
acceptance_wait_time: 10
acceptance_wait_time_max: 0
random_reauth_delay: 60
auth_timeout: 10
recon_default: 100
recon_max: 5000
recon_randomize: False
loop_interval: 60
dns_check: True
ipc_mode: ipc
tcp_pub_port: 4510
tcp_pull_port: 4511
pillar_dirs: [/srv/salt/_pillar, /srv/salt/makina-states/mc_states/pillar] 
module_dirs: [/srv/salt/_modules, /srv/salt/makina-states/mc_states/modules] 
returner_dirs: [/srv/salt/_returners, /srv/salt/makina-states/mc_states/returners] 
states_dirs: [/srv/salt/_states, /srv/salt/makina-states/mc_states/states] 
render_dirs: [/srv/salt/_renderers, /srv/salt/makina-states/mc_states/renderers] 
grain_dirs: [/srv/salt/_grains, /srv/salt/makina-states/mc_states/grains] 
grains_dirs: [/srv/salt/_grains, /srv/salt/makina-states/mc_states/grains] 
providers: {} 
cython_enable: False
renderer: yaml_jinja
failhard: False
autoload_dynamic_modules: True
clean_dynamic_modules: True
state_top: top.sls
startup_states: ''
sls_list: []
top_file: ""
file_client: remote
file_roots: {base: [/srv/salt]} 
hash_type: md5
pillar_roots: {base: [/srv/pillar]} 
ext_pillar:
    - mc_pillar: {}
open_mode: False
permissive_pki_access: False
state_verbose: True
state_output: full
master_finger: ''
multiprocessing: True
log_file: /var/log/salt/salt-minion
key_logfile: /var/log/salt/salt-minion-key
log_level: warning
log_level_logfile: info
log_datefmt: '%H:%M:%S'
log_datefmt_logfile: '%Y-%m-%d %H:%M:%S'
log_fmt_console: '[%(levelname)-8s] %(message)s'
log_fmt_logfile: '%(asctime)s,%(msecs)03.0f [%(name)-17s][%(levelname)-8s] %(message)s'
log_granular_levels: {} 
pillar_roots: {base: [/srv/pillar]} 
ext_pillar:
    - mc_pillar: {}
pillar_opts: True
update_url: False
update_restart_services: []
tcp_keepalive: True
tcp_keepalive_idle: 300
tcp_keepalive_cnt: -1
tcp_keepalive_intvl: -1
win_repo_cachefile: salt://win/repo/winrepo.p
strip_colors: False
schedule:
  __mine_interval: {function: mine.update, jid_include: true, maxrunning: 2, minutes: 60}

```

**Comment by kiorky** [comment]
the minion config.

**Comment by s0undt3ch** [comment]
Thanks.

**Comment by s0undt3ch** [comment]
Once https://github.com/saltstack/salt/pull/21164 is merged, can you please try again?

**Comment by s0undt3ch** [comment]
Fixed in #21164

**Comment by kiorky** [comment]
confirmed the fix.

**Comment by s0undt3ch** [comment]
Awesome!

**Comment by kiorky** [comment]
@s0undt3ch this does not solve the problem in the longterm toward the hardfailures.
Do we open a separate issue, or are you adding that stuff soon ?

**Comment by kiorky** [comment]
(i mean either never hardfail, even after the 'future' release, or adding an opt to only report warnings)
